test/Microsoft.Health.Fhir.Tests.Integration/Persistence/FhirStorageTests.cs

```diff
@@ -322,6 +322,36 @@ public async Task WhenDeletingAResource_ThenWeGetResourceGone()
                 () => Mediator.GetResourceAsync(new ResourceKey<Observation>(saveResult.Resource.Id)));
         }
 
+        [Fact]
+        public async Task WhenDeletingAResourceThatNeverExisted_ThenReadingTheResourceReturnsNotFound()
+        {
+            string id = "missingid";
+
+            var deletedResourceKey = await Mediator.DeleteResourceAsync(new ResourceKey("Observation", id), false);
+
+            Assert.Null(deletedResourceKey.ResourceKey.VersionId);
+
+            await Assert.ThrowsAsync<ResourceNotFoundException>(
+                () => Mediator.GetResourceAsync(new ResourceKey<Observation>(id)));
+        }
+
+        [Fact]
+        public async Task WhenDeletingAResourceForASecondTime_ThenWeDoNotGetANewVersion()
+        {
+            var saveResult = await Mediator.UpsertResourceAsync(Samples.GetJsonSample("Weight"));
+
+            var resourceKey = new ResourceKey("Observation", saveResult.Resource.Id);
```

Comment:
In tests you should also be able to use the strongly typed ResourceKey
```suggestion
            var resourceKey = new ResourceKey<Observation>(saveResult.Resource.Id);
```